core/src/plugins/filed/grpc/CMakeLists.txt

```
@@ -0,0 +1,138 @@
+    set(_REFLECTION grpc++_reflection)
+  endif()
+  if(CMAKE_CROSSCOMPILING)
+    find_program(_PROTOBUF_PROTOC protoc)
+  else()
+    set(_PROTOBUF_PROTOC $<TARGET_FILE:protobuf::protoc>)
+  endif()
+  if(TARGET gRPC::grpc++)
+    set(_GRPC_GRPCPP gRPC::grpc++)
+  else()
+    set(_GRPC_GRPCPP grpc++)
+  endif()
+  if(TARGET gRPC::grpc_cpp_plugin)
+    set(_GRPC_CPP_PLUGIN_EXECUTABLE $<TARGET_FILE:gRPC::grpc_cpp_plugin>)
+  else()
+    find_program(_GRPC_CPP_PLUGIN_EXECUTABLE grpc_cpp_plugin)
+  endif()
+
+  # Proto file
+  macro(addgrpcprotofiles protosrcs protohdrs grpcsrcs grpchdrs)
+    foreach(file ${ARGN})
+      get_filename_component(absolute "${file}" ABSOLUTE)
+      get_filename_component(path "${absolute}" PATH)
+      get_filename_component(stem "${file}" NAME_WE)
+
+      set(proto_src "${CMAKE_CURRENT_BINARY_DIR}/${stem}.pb.cc")
+      set(proto_hdr "${CMAKE_CURRENT_BINARY_DIR}/${stem}.pb.h")
+      set(grpc_src "${CMAKE_CURRENT_BINARY_DIR}/${stem}.grpc.pb.cc")
+      set(grpc_hdr "${CMAKE_CURRENT_BINARY_DIR}/${stem}.grpc.pb.h")
+        if(Protobuf_VERSION VERSION_LESS 3.15.0)
+          set(_PROTOBUF_PROTOC_EXTRA_PARAMS "--experimental_allow_proto3_optional")
+        else()
+          set(_PROTOBUF_PROTOC_EXTRA_PARAMS "")
+        endif()
+      add_custom_command(
+        OUTPUT "${proto_src}" "${proto_hdr}" "${grpc_src}" "${grpc_hdr}"
+        COMMAND
+          ${_PROTOBUF_PROTOC} ARGS --grpc_out "${CMAKE_CURRENT_BINARY_DIR}"
+          --cpp_out "${CMAKE_CURRENT_BINARY_DIR}" -I "${path}"
+          --plugin=protoc-gen-grpc="${_GRPC_CPP_PLUGIN_EXECUTABLE}"
+          ${_PROTOBUF_PROTOC_EXTRA_PARAMS}
+          "${absolute}"
+        DEPENDS "${absolute}"
+      )
+
+      list(APPEND protosrcs "${proto_src}")
+      list(APPEND protohdrs "${proto_hdr}")
+      list(APPEND grpcsrcs "${grpc_src}")
+      list(APPEND grpchdrs "${grpc_hdr}")
+    endforeach()
+  endmacro()
+
+  addgrpcprotofiles(
+    protosrcs protohdrs grpcsrcs grpchdrs "proto/plugin.proto"
+    "proto/events.proto" "proto/bareos.proto" "proto/common.proto"
+  )
+
+  add_library(STATIC grpc-proto ${protosrcs} ${protohdrs} ${grpcsrcs} ${grpchdrs})
+  target_include_directories(grpc-proto PUBLIC "${CMAKE_CURRENT_BINARY_DIR}")
+
+  target_link_libraries(
+    grpc-proto PUBLIC ${_REFLECTION} ${_GRPC_GRPCPP} ${_PROTOBUF_LIBPROTOBUF}
+  )
+  # add_library(grpc-plugin SHARED grpc.cc restore.cc config.cc database.cc) #
+  # Include generated *.pb.h files target_link_libraries( grpc-plugin
+  # ${_REFLECTION} ${_GRPC_GRPCPP} ${_PROTOBUF_LIBPROTOBUF} grpc-proto )
+  #
+  # add_executable(grpc-client gclient.cc) target_link_libraries(grpc-client
+  # grpc-proto)
+
+  add_library(grpc-fd MODULE grpc.cc grpc_impl.cc bareos_api.cc)
+  set_target_properties(
+    grpc-fd PROPERTIES PREFIX "" LIBRARY_OUTPUT_DIRECTORY ..
+  )
+  install(
+    TARGETS grpc-fd
+    DESTINATION ${plugindir}
+    COMPONENT filedaemon
+  )
+  target_link_libraries(grpc-fd bareos fmt grpc-proto)
```

```suggestion
  target_link_libraries(grpc-fd bareos fmt::fmt grpc-proto)
```
always use full qualified namespaced names for imported targets